mobile_config/src/client/carrier_service_client.rs

```diff
@@ -0,0 +1,76 @@
+use super::{call_with_retry, ClientError, Settings, CACHE_EVICTION_FREQUENCY};
+use async_trait::async_trait;
+use file_store::traits::MsgVerify;
+use helium_crypto::{Keypair, PublicKey, Sign};
+use helium_proto::{
+    services::{mobile_config, Channel},
+    Message,
+};
+use retainer::Cache;
+use std::{sync::Arc, time::Duration};
+
+#[async_trait]
+pub trait CarrierServiceVerifier {
+    type Error;
+    async fn key_to_rewardable_entity<'a>(&self, entity_id: &'a str)
+        -> Result<String, Self::Error>;
+}
+#[derive(Clone)]
+pub struct CarrierServiceClient {
+    client: mobile_config::CarrierServiceClient<Channel>,
+    signing_key: Arc<Keypair>,
+    config_pubkey: PublicKey,
+    cache: Arc<Cache<String, String>>,
```

Comment:
how frequently is this verification being called? once a day during rewards calculation or periodically? seems like we're not likely to have enough service providers on the network collecting rewards for a while such that we need a cache here